### DISCUSSION TITLE
Welcome DM using Pipedream.com and Mastodon webhooks

### DISCUSSION THREAD

**Discussion by shleeable** [mastodon/mastodon#23063] [Show and tell]
I've got the `account.approved` webhook to trigger a python script every new account.

## instructions
1) Signup to pipedream.com and setup a new webhook source for the signup event.

2) Create a webhook in mastodon for the `account.approved` event using the pipedream URL.
![image](https://user-images.githubusercontent.com/17537000/212207783-8d69204b-8b66-4912-bc98-ebb465b0faea.png)

3) Create a new application as the account you went to send from with `write:statuses`.
![image](https://user-images.githubusercontent.com/17537000/212207400-e25d5c7b-2573-43ff-8545-fde1e674643b.png)

4) Paste the code I pasted and update the access token in the code.

5) update the welcome message 

6) Deploy

```
#!/usr/bin/env python

import requests

def handler(pd: "pipedream"):
  username = pd.steps["trigger"]["event"]["body"]["object"]["username"]
  # Mastodon information
  token = "XXXXXXX"
  postURL = 'https://aus.social/api/v1/statuses'

  message = '''Hello new neighbour, and welcome to Aus.Social!

Feel free to write an #Introduction / #introductions post. This hashtag is followed by people in our community.\n\nMastodon has a few teething issues, but aus.social is built to be a long term stable home for us all. If you experience any problems, please message me.

Asking questions is okay and encouraged! If you have any issues with people in the fediverse, please report them. Our mods are here to help.

Enjoy, and this place gets better with friends :)

- The administrator and team'''

  toot = '''@{} {}'''.format(username, message)

  # Send the toot.
  header = {'Authorization': 'Bearer {}'.format(token)}
  payload = {'status': toot, 'visibility': "direct"}
  r = requests.post(postURL, data=payload, headers=header)

  print(r.json()['uri'])
```

![image](https://user-images.githubusercontent.com/17537000/212067537-afee16de-94db-44dd-9ed7-f40beb7d5194.png)

Results in a DM to every new user

![image](https://user-images.githubusercontent.com/17537000/212066857-35a80eb0-6f98-4e88-9e60-1f540121c795.png)

**Comment by mrmartineau** [comment]
@shleeable how do you setup the webhook? Is it in the Mastodon web client? I can't see it in mine. Did it get added in a particular version?

**Reply by andypiper** [reply]
This is under the Administration -> Webhooks section of the Mastodon preferences in the web UI, for server owners.

**Reply by mrmartineau** [reply]
ah, it's for server owners, I didn't realise that. Thanks
